src/components/AccountIconChooser.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+// @flow
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { View, Text, StyleSheet, ListView, TouchableOpacity } from 'react-native';
+import { brainWalletAddress, words } from '../util/native';
+import colors from '../colors';
+import Card from './Card';
+import AccountIcon from './AccountIcon';
+
+export default class AccountIconChooser extends Component<{
+  value: string,
+  onChange: () => any
+}> {
+  static propTypes = {
+    value: PropTypes.string.isRequired,
+    onChange: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+    this.icons = [];
+    const iconsDS = new ListView.DataSource({ rowHasChanged: (r1, r2) => true });
+    this.state = { iconsDS };
+  }
+
+  refreshIcons = async () => {
+    try {
+      this.icons = [
+        ...this.icons,
+        ...(await Promise.all(
+          Array(10)
+            .join(' ')
+            .split(' ')
+            .map(async () => {
+              const seed = await words();
+              return {
+                seed,
+                address: (await brainWalletAddress(seed)).toLowerCase()
+              };
+            })
+        ))
+      ];
+      this.setState({ iconsDS: this.state.iconsDS.cloneWithRows(this.icons) });
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  componentDidMount() {
+    this.refreshIcons();
+  }
+
+  render() {
+    const { value, onChange } = this.props;
+
+    return (
+      <View style={styles.body}>
+        <ListView
+          style={styles.icons}
+          dataSource={this.state.iconsDS}
+          horizontal={true}
+          renderRow={({ address, seed }, sectionID: number, rowID: number, highlightRow) => {
+            const selected = address === value;
+            const style = [styles.icon];
+            return (
+              <TouchableOpacity
+                style={[styles.iconBorder, address === value ? styles.selected : {}]}
+                onPress={() => this.props.onChange({ address, seed })}
```

Comment:
Same thing as a bit above: it's recommended to use `onPress={this.handlePress}`, and `handlePress` is a method that does `this.props.onChange({ address, seed }).`. This will avoid creating a new function on every render (less work for the garbage collector).

Reply:
So, if I really want to keep these function inline, doest the following do the trick, how do you think?  

```
onChangeText={function some_name (name) { accounts.updateNew({ name })}}
```